src/audio_worklet_node.rs

```diff
@@ -0,0 +1,260 @@
+        item.2.into_iter().for_each(|(name, ptr, len)| {
+            let val = float_buffer_to_js(&env, ptr, len);
+            params.set_named_property(&name, val).unwrap()
+        });
+
+        let js_ret: JsUnknown = process.apply3(proc, inputs, outputs, params)?;
+        let _ret = js_ret.coerce_to_bool()?.get_value()?;
+    }
+    env.get_undefined()
+}
+
+pub(crate) struct NapiAudioWorkletNode(AudioWorkletNode);
+
+impl NapiAudioWorkletNode {
+    pub fn create_js_class(env: &Env) -> Result<JsFunction> {
+        let interface = audio_node_interface![];
+
+        env.define_class("AudioWorkletNode", constructor, &interface)
+    }
+
+    pub fn unwrap(&self) -> &AudioWorkletNode {
+        &self.0
+    }
+}
+
+#[js_function(3)]
+fn constructor(ctx: CallContext) -> Result<JsUndefined> {
+    let mut js_this = ctx.this_unchecked::<JsObject>();
+
+    let js_audio_context = ctx.get::<JsObject>(0)?;
+
+    // --------------------------------------------------------
+    // Create AudioBufferSourceOptions object
+    // --------------------------------------------------------
+    let options: AudioWorkletNodeOptions<()> = Default::default();
+    let send = send_recv_pair().lock().unwrap().0.take().unwrap();
+
+    // Remap the constructor options to include our processor options
+    let AudioWorkletNodeOptions {
+        number_of_inputs,
+        number_of_outputs,
+        output_channel_count,
+        parameter_data,
+        processor_options: _processor_options,
+        audio_node_options,
+    } = options;
+    let options = AudioWorkletNodeOptions {
+        number_of_inputs,
+        number_of_outputs,
+        output_channel_count,
+        parameter_data,
+        audio_node_options,
+        processor_options: send,
+    };
+
+    // --------------------------------------------------------
+    // Create native AudioWorkletNode
+    // --------------------------------------------------------
+    let audio_context_name =
+        js_audio_context.get_named_property::<JsString>("Symbol.toStringTag")?;
+    let audio_context_utf8_name = audio_context_name.into_utf8()?.into_owned()?;
+    let audio_context_str = &audio_context_utf8_name[..];
+
+    let native_node = match audio_context_str {
+        "AudioContext" => {
+            let napi_audio_context = ctx.env.unwrap::<NapiAudioContext>(&js_audio_context)?;
+            let audio_context = napi_audio_context.unwrap();
+            AudioWorkletNode::new::<NapiAudioWorkletProcessor>(audio_context, options)
+        }
+        "OfflineAudioContext" => {
+            let napi_audio_context = ctx
+                .env
+                .unwrap::<NapiOfflineAudioContext>(&js_audio_context)?;
+            let audio_context = napi_audio_context.unwrap();
+            AudioWorkletNode::new::<NapiAudioWorkletProcessor>(audio_context, options)
+        }
+        &_ => panic!("not supported"),
+    };
+
+    /* TODO expose parameters in JS
+    dbg!(native_node.parameters());
```

Comment:
@b-ma I could maybe use your help to expose the AudioParams to javascript here.
If you uncomment the rust lines and run the example you will see that the bitcrusher now works with automations, but obviously we should be able to control the params from the JS side.

Reply:
Ok sure, I will have a look

Just want to fix this `index.js|ts` files issue first